src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/InfrastructureFormula.kt

```diff
@@ -31,8 +31,16 @@ class InfrastructureFormula<out T : VirtualUsers>(
     private val investment: Investment,
     private val jiraFormula: JiraFormula,
     private val virtualUsersFormula: VirtualUsersFormula<T>,
-    private val aws: Aws
+    private val aws: Aws,
+    private var preProvisionedNetwork: Network?
 ) {
+    constructor(
```

Comment:
Multiple constructors mean it's time to go for a builder instead. See [Java API checklist 3.6.12](https://github.com/wyrzyk/java-api-design-checklist/blob/d0dbffa525007f320892b382a3e477702707cd93/CHECKLIST.md#3612-consider-a-builder-when-faced-with-many-constructor-parameters). Then, the new constructor should be `private` to shield maintainers from future compatibility pain.

Reply:
added